File: netty/src/main/java/cn/elmi/sample/netty/chat/ChatClient.java
package cn.elmi.sample.netty.chat;

import java.io.BufferedReader;
import java.io.InputStreamReader;

import io.netty.bootstrap.Bootstrap;
import io.netty.channel.Channel;
import io.netty.channel.EventLoopGroup;
import io.netty.channel.nio.NioEventLoopGroup;
import io.netty.channel.socket.nio.NioSocketChannel;
import lombok.extern.slf4j.Slf4j;

/**
 * @version 1.0
 * @authur Arthur
 */
@Slf4j
public class ChatClient {

    public static void main(String[] args) {
        EventLoopGroup group = new NioEventLoopGroup();

        try {
            Bootstrap bootstrap = new Bootstrap().group(group)
                    .channel(NioSocketChannel.class)
                    .handler(new ChatClientInitializer());

            Channel channel = bootstrap.connect("localhost", 8800).sync().channel();
            BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

            for (; ; ) {
                channel.writeAndFlush(in.readLine() + "\r\n");
            }


//            channelFuture.channel().closeFuture().sync();
        } catch (Exception e) {
            log.error(e.getMessage());
        } finally {
            group.shutdownGracefully();
        }

    }


}

File: netty/src/main/kotlin/cn/elmi/sample/netty/chat/ChatClient.kt
package cn.elmi.sample.netty.chat

import io.netty.bootstrap.Bootstrap
import io.netty.channel.EventLoopGroup
import io.netty.channel.nio.NioEventLoopGroup
import io.netty.channel.socket.nio.NioSocketChannel
import java.io.BufferedReader
import java.io.InputStreamReader

/**
 * @version 1.0
 * @authur Arthur
 */
object ChatClient {
    @JvmStatic
    fun main(args: Array<String>) {
        val group: EventLoopGroup = NioEventLoopGroup()
        try {
            val bootstrap = Bootstrap().group(group)
                .channel(NioSocketChannel::class.java)
                .handler(ChatClientInitializer())
            val channel = bootstrap.connect("localhost", 8800).sync().channel()
            val `in` = BufferedReader(InputStreamReader(System.`in`))
            while (true) {
                channel.writeAndFlush(
                    """
    ${`in`.readLine()}
    
    """.trimIndent()
                )
            }


//            channelFuture.channel().closeFuture().sync();
        } catch (e: Exception) {
//            log.error(e.message)
        } finally {
            group.shutdownGracefully()
        }
    }
}